dataregistry-extension/src/csvviewer/activate.ts

```diff
@@ -0,0 +1,129 @@
+/**
```

Comment:
Instead of `activate.ts` maybe have `converters.ts` file that exports function from `registry` to array of converters.

Reply:
My motivation: (1) 1 file, 1 concern and (2) testing.
​
In short, a guiding principle in Node.js development is to create files (modules) which export one thing (i.e., have a single concern), in order to encourage readability and testability.
​
If a file contains a single export, no need to track down a file's exports, which, if a file is long and not structured, can be tedious and time consuming.
​
And if a file has a single export, encourages writing tests which import that single export and directly test an API surface area. Often when everything is in long files, we see correspondingly long test files with correspondingly increased difficulty in achieving 100% code coverage. By having single file tests, easier to test using mocks via dependency injection (classic is proxyquire to proxy imported dependencies) and, as tests are documentation, easier to see how exported functionality should behave.
​
For example, take `activate.ts`. The file exports a single function which accepts two arguments and whose intent is to register "converters". When testing, we can import the sole export from this individual file, and, as Jupyter uses dependency injection, we can mock the second argument with an object containing a single method: `addConverter`, testing whether particular converters are registered, etc.
​
Similarly, I moved the converters to separate files based on the same idea. We should be able to test a particular converter, say, the one generated by export from `text_to_csv`, to ensure proper conversion behavior, such as the conversion of text "observable" to a CSV "observable", etc. If everything is in a single file and all the logic tightly coupled, this becomes rather difficult to test, especially thoroughly.
​
In a sense, each file having a single concern allows for, perhaps arguably, easier enforcement of API contracts.
​
Single-file-single-concern "style" can be found, to some degree, in Node.js core, and more so in Node.js modules written with the file system in mind and not, say, like jQuery, as a single file to be downloaded by a browser.
​
This "style" takes inspiration from the Unix philosophy in the sense that each module should do only one thing and do that thing well. Often, we think of this at the package level, but applies equally to the module/file layer.